### PR TITLE
Include eventv1.MetaTokenKey on event rate limiting key calculation

### PR DESCRIPTION
**Pull request by matheuscscp**
Part of fixing https://github.com/fluxcd/helm-controller/issues/678

Supersedes https://github.com/fluxcd/notification-controller/pull/518

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/Azure/azure-event-hubs-go/v3 v3.5.0
 	github.com/containrrr/shoutrrr v0.7.1
 	github.com/fluxcd/notification-controller/api v1.0.0-rc.3
-	github.com/fluxcd/pkg/apis/event v0.4.1
+	github.com/fluxcd/pkg/apis/event v0.4.2-0.20230516071635-f4cf2dc1a8b0
```
Comment (hiddeco) [inline go.mod:14]: We are awaiting a controller-runtime release to pair with an update to Kubernetes 1.27.x, which would probably be best to land before tagging this.

Comment (matheuscscp) [reply go.mod:14]: When is that expected to happen?

Comment (hiddeco) [reply go.mod:14]: Hopefully this week, as it is also blocking Kustomize v5 😬

Comment (hiddeco) [reply go.mod:14]: This has seen a release now.

**Review by stefanprodan — APPROVED** [review]
LGTM

**Pull request merged**